### PR TITLE
Core: fixing configuration file resolution algorithm

### PR DESCRIPTION
**Pull request by amnonshmsft**

### CODE REVIEW

File: core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java [hunk 1]
```diff
@@ -333,75 +331,57 @@ private boolean setChannel(ChannelXmlElement channelXmlElement, TelemetryConfigu
...
-            }
+        return configurationFile;
+    }
 
-            String configurationFile = getConfigurationAbsolutePath(configurationPath);
-            if (configurationFile != null) {
-                return configurationFile;
-            } else {
-                checkedUrls.add(configurationPath);
+    private String getConfigurationFromDefaultClassLoader() {
```
Comment (gupele) [inline core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java]: should be 'getConfigurationFromCurrentClassLoader'

File: core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java [hunk 2]
```diff
@@ -333,75 +331,57 @@ private boolean setChannel(ChannelXmlElement channelXmlElement, TelemetryConfigu
...
-                } else {
-                    continue;
-                }
-            } else {
-                if (!configurationPath.endsWith("/")) {
-                    configurationPath += '/';
-                }
-            }
+        if (configurationFile != null) {
+            InternalLogger.INSTANCE.logAlways(InternalLogger.LoggingLevel.INFO, "Configuration file found in: '%s'", configurationFile);
```
Comment (gupele) [inline core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java]: "Configuration file has been successfully found in:......." this is what is written in the Blob

File: core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java [hunk 3]
```diff
@@ -333,75 +331,57 @@ private boolean setChannel(ChannelXmlElement channelXmlElement, TelemetryConfigu
...
 
-        // If not found as a resource, trying to load from the class path
-        return getConfFromClassPath(classLoader);
-    }
-
-    private String getConfFromClassPath(ClassLoader classLoader) {
-        if (!(classLoader instanceof URLClassLoader)) {
-            return null;
+        // If still not found try to get it from the class path
+        if (configurationFile == null) {
```
Comment (gupele) [inline core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java]: This should be in first check of the 'if (configurationFile == null)' this is activated only if the second one produces null

File: core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java [hunk 4]
```diff
@@ -85,7 +83,7 @@ public final void initialize(TelemetryConfiguration configuration) {
                 return;
             }
 
-            ApplicationInsightsXmlConfiguration  applicationInsightsConfig = builder.build(configurationFile);
+            ApplicationInsightsXmlConfiguration applicationInsightsConfig = builder.build(configurationFile);
```
Comment (gupele) [inline core/src/main/java/com/microsoft/applicationinsights/internal/config/TelemetryConfigurationFactory.java]: I think all this logic should be placed in a dedicated class, which is easier to test

**Comment by amnonshmsft** [comment]
Code submitted by Guy in another PR.

**Pull request closed without merging**